x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsProcessManager.java

```diff
@@ -145,4 +154,20 @@ private AnalyticsProcessConfig createProcessConfig(DataFrameAnalyticsConfig conf
                 new ByteSizeValue(1, ByteSizeUnit.GB), 1, dataFrameAnalyses.get(0));
         return processConfig;
     }
+
+    @Nullable
+    public Integer getProgressPercentage(long allocationId) {
+        ProcessContext processContext = processContextByAllocation.get(allocationId);
+        return processContext == null ? null : processContext.progressPercentage.get();
+    }
+
+    static class ProcessContext {
+
+        private final AtomicInteger progressPercentage = new AtomicInteger(0);
+
+        void setProgress(double progress) {
+            progressPercentage.set((int) Math.round(progress * 100));
```

Comment:
It should probably be `floor`, not `round`.  Otherwise for a long analysis progress of 100% will be reported when there could still be minutes of processing left to do.